elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/json/ToElasticsearch.java

```diff
@@ -0,0 +1,287 @@
+		}
+
+		if ( shoulds.size() > 1 ) {
+			clauses.add( "should", shoulds );
+		}
+		else if ( shoulds.size() == 1 ) {
+			clauses.add( "should", shoulds.iterator().next() );
+		}
+
+		if ( mustNots.size() > 1 ) {
+			clauses.add( "must_not", mustNots );
+		}
+		else if ( mustNots.size() == 1 ) {
+			clauses.add( "must_not", mustNots.iterator().next() );
+		}
+
+		if ( filters.size() > 1 ) {
+			clauses.add( "filter", filters );
+		}
+		else if ( filters.size() == 1 ) {
+			clauses.add( "filter", filters.iterator().next() );
+		}
+
+		JsonObject bool = new JsonObject();
+		bool.add( "bool", clauses );
+		return bool;
+	}
+
+	private static JsonObject convertTermQuery(TermQuery termQuery) {
+		String field = termQuery.getTerm().field();
+
+		JsonObject matchQuery = JsonBuilder.object()
+				.add( "match",
+						JsonBuilder.object().addProperty( field, termQuery.getTerm().text() ) )
+				.build();
+
+		return wrapQueryForNestedIfRequired( field, matchQuery );
+	}
+
+	private static JsonObject convertWildcardQuery(WildcardQuery query) {
+		String field = query.getTerm().field();
+
+		JsonObject wildcardQuery = JsonBuilder.object()
+				.add( "wildcard",
+						JsonBuilder.object().addProperty( field, query.getTerm().text() ) )
+				.build();
+
+		return wrapQueryForNestedIfRequired( field, wildcardQuery );
+	}
+
+	private static JsonObject convertTermRangeQuery(TermRangeQuery query) {
+		JsonObject interval = new JsonObject();
+
+		if ( query.getLowerTerm() != null ) {
+			interval.addProperty( query.includesLower() ? "gte" : "gt", query.getLowerTerm().utf8ToString() );
+		}
+		if ( query.getUpperTerm() != null ) {
+			interval.addProperty( query.includesUpper() ? "lte" : "lt", query.getUpperTerm().utf8ToString() );
+		}
+
+		JsonObject range = JsonBuilder.object().add( "range",
+						JsonBuilder.object().add( query.getField(), interval ))
+				.build();
+
+		return wrapQueryForNestedIfRequired( query.getField(), range);
+	}
+
+	private static JsonObject convertNumericRangeQuery(NumericRangeQuery<?> query) {
+		JsonObject interval = new JsonObject();
+		if ( query.getMin() != null ) {
+			interval.addProperty( query.includesMin() ? "gte" : "gt", query.getMin() );
+		}
+		if ( query.getMax() != null ) {
+			interval.addProperty( query.includesMax() ? "lte" : "lt", query.getMax() );
+		}
+
+		JsonObject range = JsonBuilder.object().add( "range",
+						JsonBuilder.object().add( query.getField(), interval ))
+				.build();
+
+		return wrapQueryForNestedIfRequired( query.getField(), range);
+	}
+
+	private static JsonObject wrapQueryForNestedIfRequired(String field, JsonObject query) {
+		if ( !isFieldNested( field ) ) {
+			return query;
+		}
+		String path = getFieldNestedPath( field );
+
+		return JsonBuilder.object().add( "nested",
+				JsonBuilder.object()
+						.addProperty( "path", path )
+						.add( "query", query ) )
+				.build();
+	}
+
+	public static JsonObject fromLuceneFilter(Filter luceneFilter) {
```

Comment:
Haven't tested it yet but I think so. At least for the filters built from a query.